providers/async_registry.py

```
@@ -0,0 +1,350 @@
+        instance._async_providers.pop(provider_type, None)
+        instance._async_initialized_providers.pop(provider_type, None)
+
+    def get_connection_stats(self) -> dict:
+        """
+        Return statistics about initialized synchronous and asynchronous providers, including detailed connection stats for each async provider if available.
+
+        Returns:
+            dict: A dictionary containing counts of initialized sync and async providers, and per-provider connection statistics for async providers that support it.
+        """
+        stats = {
+            "sync_providers": len(self._initialized_providers),
```

Pylint (3.3.7)

[error] Instance of 'AsyncModelProviderRegistry' has no '_initialized_providers' member

(E1101)